### ISSUE TITLE
reading-time v2

### ISSUE THREAD

**Issue by Josh-Cena** [ngryman/reading-time#38]
Here are a few changes we can make in v2 (most of which @ngryman has also mentioned), roughly in order of decreasing importance:

0. Refactor the API to expose two separate functions;
1. Remove `text` from the function return;
2. ~~Remove the badly named `IOptions` export;~~
3. ~~Migrate to TypeScript completely;~~
4. Prefer ESM to CJS (once we have TypeScript this is just a matter of setting the transpilation target);
5. ~~Use a class-based implementation of `readingTimeStream`;~~
6. ~~Change the testing library from mocha to Jest.~~

@ngryman If you have limited availability, I can be here to help.

**Comment by ngryman** [comment]
All of this sounds great to me, thanks!

I finally moved so I should be more available starting next week. I should be able to help here.

Ref #18 for stuff related to the API changes.

**Comment by Josh-Cena** [comment]
Number 2, 3 and 5 will be resolved in #40.

**Comment by sanderjson** [comment]
> Just jumping in to say that, with my current setup, `v1.4.0` worked fine without any changes, but updating to `v1.5.0` required me to add the following to my webpack config:
> 
> ```js

> node: {
>   global: true
> },
> resolve: {
>   fallback: {
>     stream: 'stream-browserify'
>   }
> },
> plugins: [
>   new webpack.ProvidePlugin({
>     process: 'process/browser',
>     Buffer: ['buffer', 'Buffer']
>   })
> ]
> ```
> 
> and had to install `stream-browserify`.
> 
> The problematic line was `const Transform = require('stream').Transform`, but this line is also present in `v1.4.0`. This is a node core package, so it is expected for some kind of polyfill to be needed, but for some strange reason, `v1.4.0` just works without any particular configuration change. 🤷‍♂️

---

Not sure if this is the same issue but I had problems with 1.5 on Vite. Using 1.4 for now.
```
util.inherits is not a function
TypeError: util.inherits is not a function
```

**Comment by Josh-Cena** [comment]
@sanderjson Yes, your problem is the same as the previous one. `util` is a Node library and is not polyfilled by Vite. You can either use 1.4, use 2.0.0-1 (which has a `browser` field in package.json), or import from `reading-time/lib/reading-time` instead.

**Comment by honwhy** [comment]
any updates ?

**Comment by SalahAdDin** [comment]
Is this being maintained?
